### PR TITLE
Fix build_sysimg.jl  on Windows when linking failure occurs

### PR DESCRIPTION
**Pull request by musm**
In case linking fails, load the previous sysimg_file so that Julia can actually start  (reported issue with a user having the wrong gcc from WinRPM and then could not reload julia because the `sysimg_file` was not renamed back).

### CODE REVIEW

**Comment by simonbyrne** [comment]
wouldn't it be safe to generate a new file, and only overwrite it if it was successful?

**Comment by musm** [comment]
Not sure exactly what you mean? Right now it renames the currently loaded dll file since you can't overwrite it.   This PR just makes sure we rename it back if the linking fails

**Comment by simonbyrne** [comment]
Instead of:
1. rename current -> old
2. build to current
3. if failed: rename old -> current

Do:
1. build to temp file
2. if successful: rename current -> old, temp -> current

**Comment by musm** [comment]
done

**Comment by musm** [comment]
bump

**Review by simonbyrne — APPROVED** [review]

**Comment by musm** [comment]
Updated in case of error. Also please don't forget to squash this.

**Comment by musm** [comment]
bumping again, anything else need to be done here?

**Comment by StefanKarpinski** [comment]
@simonbyrne: you've already reviewed this. If it still looks good to you, please merge.

File: contrib/build_sysimg.jl [hunk 1]
```diff
@@ -157,10 +157,13 @@ function link_sysimg(sysimg_path=nothing, cc=find_system_compiler(), debug=false
 
     sysimg_file = "$sysimg_path.$(Libdl.dlext)"
     info("Linking sys.$(Libdl.dlext)")
-    # Windows has difficulties overwriting a file in use so we first rename it
+    info(`$cc $FLAGS -o $sysimg_file $sysimg_path.o`)
```
Comment (tkelman) [inline contrib/build_sysimg.jl]: none of the other infos are on backtick cmds

**Pull request merged**